extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/OidcBuildStep.java

```diff
@@ -323,13 +326,21 @@ private static boolean isInjected(BeanRegistrationPhaseBuildItem beanRegistratio
             DotName withoutQualifier) {
         for (InjectionPointInfo injectionPoint : beanRegistrationPhaseBuildItem.getInjectionPoints()) {
             if (requiredType.equals(injectionPoint.getRequiredType().name())
+                    && isApplicationPackage(injectionPoint.getTargetInfo())
                     && (withoutQualifier == null || injectionPoint.getRequiredQualifier(withoutQualifier) == null)) {
+                LOG.debugf("%s injection point: %s", requiredType.toString(), injectionPoint.getTargetInfo());
                 return true;
             }
         }
         return false;
     }
 
+    private static boolean isApplicationPackage(String injectionPointTargetInfo) {
+        return injectionPointTargetInfo != null
+                && !injectionPointTargetInfo.startsWith(QUARKUS_TOKEN_PROPAGATION_PACKAGE)
```

Comment:
The OIDC Token propagation line basically says that we register filters that are not actually used while we have all build time information:

1. We can detect when the filter is needed
2. If the `JsonWebToken` is actually injected inside bean used by the application (by actively used JAX-RS filter), then it should use verified JWT?

As for SmallRye JWT package unfortunately I don't know it well enough, but to have unused beans registered by default surprises me.

Reply:
@michalvavrik 

>The OIDC Token propagation line basically says that we register filters that are not actually used while we have all build time information...We can detect when the filter is needed

This is only available in the Resteasy Classic based optional filter which as far as I know noone has used, this is why I did not make the same feature available in the reactive propagation filter. That feature is about using a JWT build API to construct a new JWT token from the coming JWT token and then resigning it with the new key, setting a new audience. But users never used it because the token propagation supports  the token exchange grants.
So this check is just to avoid that unused feature interfering, IMHO it is not worth the effort and start checking at the quarkus-oidc level if that filter is used given that filter is also expected to work with smallrye-jwt.

>If the JsonWebToken is actually injected inside bean used by the application (by actively used JAX-RS filter), then it should use verified JWT?

If it is meant to be propagated then not really, we won't use for any security decisions locally and users won't access it directly.

We are talking about this feature: https://quarkus.io/guides/security-openid-connect-client-reference#restclient-jsonwebtokenrequestfilter

As I said I haven' seen any evidence it being used, instead the exchange token grants are used to set a new audience, resign etc. And like I said, in cases where it is not enforced users can just enable with the property. But if resteasy easy client users will use that extension to propagate Google binary access token, it will cause a failure
because we will detect `JsonWebToken` - so if someone, theoretically at least, uses that feature, and wants to have it verified they can enable the verification, I suppose I can update the migration guide.

>As for SmallRye JWT package unfortunately I don't know it well enough, but to have unused beans registered by default surprises me.

That producer is probably 7 years old, it may have been myself or Scott who coded it, fair to say it is not the best CDI code written but we just should not let in interfere in this feature

Reply:
Copy that. Thanks for in detail explanation.